Sources/SentryCrash/Recording/SentryCrashBinaryImageCache.c

```diff
@@ -150,6 +151,42 @@ sentrycrashbic_iterateOverImages(sentrycrashbic_imageIteratorCallback callback,
     }
 }
 
+/** Check if dyld should be added to the binary image cache.
+ *
+ * Since Apple no longer includes dyld in the images listed by _dyld_image_count and related
+ * functions, we need to check if dyld is already present in our cache before adding it.
+ *
+ * @return true if dyld is not found in the loaded images and should be added to the cache,
+ *         false if dyld is already present in the loaded images.
+ */
+bool
+sentrycrashbic_shouldAddDyld(void)
+{
+    // dyld is different from libdyld.dylib, the later contains the public API
+    // while the former contains the runtime information
```

Comment:
I don't quite understand the distinction between public API and runtime information, could you elaborate?